bricks/_common/mpconfigport.h

```diff
@@ -40,7 +40,7 @@
 #define MICROPY_ALLOC_PATH_MAX                  (256)
 #define MICROPY_ALLOC_PARSE_CHUNK_INIT          (16)
 #define MICROPY_EMIT_X64                        (0)
-#define MICROPY_EMIT_THUMB                      (0)
+#define MICROPY_EMIT_THUMB                      (PYBRICKS_OPT_EXTERNAL_MOD)
```

Comment:
I think `EXTERNAL` has a different meaning already in MicroPython (e.g. see `MICROPY_ENABLE_EXTERNAL_IMPORT`). Perhaps a better name is `PYBRICKS_OPT_NATIVE_MOD`.

Reply:
This should also depend on `__thumb2__` in case this config file is used on a different or older arch.
```suggestion
#define MICROPY_EMIT_THUMB                      (PYBRICKS_OPT_NATIVE_MOD && __thumb2__)
```

Reply:
Would you not simply set it to false on any device with an older CPU?

Reply:
For older Arm archs, sure, makes sense. But it still makes sense to do this for compatibility with non-Arm: 

```c
#define MICROPY_EMIT_X64                        (PYBRICKS_OPT_NATIVE_MOD && __x86_64__)
#define MICROPY_EMIT_THUMB                      (PYBRICKS_OPT_NATIVE_MOD && __thumb2__)
```

Reply:
Alright